### ISSUE TITLE
Application icon

### ISSUE THREAD

**Issue by selurvedu** [sm64pc/sm64pc#244]
I [made](https://aur.archlinux.org/packages/sm64pc-git/#comment-746747) a [desktop entry](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) for sm64pc, but it does not have a launcher icon yet because there is no icon in the source repo. Windows exe builds probably lack it too. Please consider adding it.

**Comment by fgsfdsfgs** [comment]
Having an icon in the repo would mean it would have to be one that is not made of copyrighted material. Of course it might be a good idea to rip it from the ROM along with the other graphics (e.g. the Mario head from the lives counter), but we might also need a bigger version, so a custom icon would probably be better in the long run.

**Comment by selurvedu** [comment]
> copyrighted material

Like this image? https://github.com/sm64pc/sm64pc/blob/89e86908578d2fdc424f9f47d91b1d82948eaa59/doxygen/logo.png :-)

> Having an icon in the repo would mean it would have to be one that is not made of copyrighted material.
> We might also need a bigger version, so a custom icon would probably be better in the long run.

I think there is no other option for making a proper icon, besides having no icon at all, since there is no such thing as "game icon" in N64. A substitute ripped from the ROM, like Mario head icon you mentioned, can be used in the meanwhile until someone makes a proper icon.

**Comment by fgsfdsfgs** [comment]
Now you've done it. Nintendo's going to fucking assassinate us.
In any case, the less such material's in the repo, the better, especially considering that the icon will also be in repos like AUR or whatever.
If someone wants to make an icon for this, they're welcome. Graphics design is definitely not my passion.

**Comment by wabberz** [comment]
I didn't know this code had a license, much less MIT...

**Comment by vrmiguel** [comment]
> I didn't know this code had a license, much less MIT...

Yeah, it's not MIT (or any other license). The guy who put it on AUR slapped a MIT license on it for some reason.

https://aur.archlinux.org/packages/sm64pc-git/

**Comment by selurvedu** [comment]
> Now you've done it. Nintendo's going to fucking assassinate us.

Haha, I thought it would be better if the project collaborators got aware of its existence from my comment rather than from a DMCA takedown notice. :-)

> In any case, the less such material's in the repo, the better

Yes, that's why I decided to let everyone here know about the file which probably needs to be removed.

**Comment by selurvedu** [comment]
@wabberz @vrmiguel 

> > I didn't know this code had a license, much less MIT...
> 
> Yeah, it's not MIT (or any other license). The guy who put it on AUR slapped a MIT license on it for some reason.
> 
> https://aur.archlinux.org/packages/sm64pc-git/

The guy [said](https://aur.archlinux.org/packages/sm64pc-git#comment-744927):

> @ Hengi they have no LICENSE file, nor does n64decomp/sm64, but they are based on tools that are MIT, therefore if not mentioned, MIT by default, most importantly, if it has no license, WHO CARES? It litteraly means i found code on street, i then can license it. #ImNoLawyer

Honestly, I think that's BS, but since I'm neither a maintainer of that AUR package nor a person who has a right to decide how sm64pc is licensed, I simply disregarded that.

**Comment by wabberz** [comment]
Putting a wrong license on code you don't own is a surefire way of getting your package removed.

**Comment by selurvedu** [comment]
@wabberz which license is considered right for sm64pc? I guess nobody has the answer.

[This arch wiki page](https://wiki.archlinux.org/index.php/PKGBUILD#license) suggests:

> If after researching the issue no license can be determined, [PKGBUILD.proto](https://projects.archlinux.org/pacman.git/tree/proto/PKGBUILD.proto) suggests using `unknown`. However, upstream should be contacted about the conditions under which the software is (and is not) available.

Would `unknown` be appropriate for this project?

**Comment by wabberz** [comment]
The legal situation is not entirely clear here. You could argue that the team that decompiled the code owns the rights to their "interpretation of the binary in C" and that this team owns the rights to the changes made from said decompilation. Nintendo obviously owns the rights to all assets.
It's probably best to set it to "unknown", since none of us is a lawyer to figure this out. I only translate silly lawyer games...

**Comment by selurvedu** [comment]
@wabberz what you wrote is exactly my understanding of the situation. Thank you for confirming it.

I'll ask the AUR maintainer to replace "MIT" with "unknown" if everyone else is okay with that (I'm not sure who to cc here).

**Comment by enigma9o7** [comment]
It would be cool if it extracted something from the rom (like marios head) during build and left it as a png (or similar) for use as an icon... but is that possible as part of building?  or is that just something completely different, its not like one can't find icons online already, but if something popped out during build, that'd be even easier.

**Comment by fgsfdsfgs** [comment]
The Mario head already gets saved somewhere in `textures/segment2/` as part of the asset extraction process. Problem is, it's just 16x16 and upscaling that would probably look like arse.

**Comment by enigma9o7** [comment]
Sweet!  Thanks!  
segment2.05A00.rgba16
That's good enough for an icon for a linux desktop file menu entry if I don't want to look elsewhere.

**Comment by dgeneratioN** [comment]
i think segment2.10458.ia8.png also looks decent and its 64x64

**Comment by Patjwmiller** [comment]
I feel like most people on here already added a titlebar and prog icon just for their own esthetics. I did at least.
